ibis/expr/datatypes/core.py

```diff
@@ -1720,7 +1738,13 @@ def can_cast_floats(
 @castable.register(Decimal, Decimal)
 def can_cast_decimals(source: Decimal, target: Decimal, **kwargs) -> bool:
     return (
-        target.precision >= source.precision and target.scale >= source.scale
+        (tgt_prec := target.precision) is None
```

Comment:
Can you remove the walrus assignment? It makes the condition pretty hard to read.